### PR TITLE
feat: add logger package

### PR DESCRIPTION
**Pull request by md0x**
Changes proposed in this PR:
- This PR adds a standalone logger package with minimal dependencies. It addresses compatibility issues encountered when using financial-templates with ethers v6.x

### CODE REVIEW

**Review by daywiss — COMMENTED** [review]
maybe i missed this discussion but why is this a whole new package?

**Comment by md0x** [comment]
> maybe i missed this discussion but why is this a whole new package?

Hey @daywiss, we cannot import the Logger we use everywhere in the bots from [financial-templates](https://github.com/UMAprotocol/protocol/blob/master/packages/financial-templates-lib/package.json#L10) in [oev-backrunner](https://github.com/UMAprotocol/oev-backrunner/) because it has ethers v5.x nested in some dependencies and in oev-backrunner we use ethers v.6.x.

The easiest and cleanest was to create a new package with the logger

File: packages/logger/package.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    "minimist": "^1.2.0",
+    "moment": "^2.29.1",
+    "moment-timezone": "^0.5.33",
+    "node-fetch": "^2.6.0",
+    "node-pagerduty": "^1.2.0",
+    "superstruct": "^1.0.3",
+    "web3": "^1.6.0",
+    "winston": "^3.2.1",
+    "winston-transport": "^4.3.0"
+  },
```
Comment (mrice32) [inline packages/logger/package.json:27]: Do we really need all these dependencies? We should ensure that everything in dependencies is imported somewhere in the _production code_ (note: dev dependencies can be imported in tests or used by yarn scripts/development).

File: packages/logger/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# @uma/logger
```
Comment (mrice32) [inline packages/logger/README.md:1]: Great docs!

File: packages/logger/src/logger/ConsoleTransport.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+// This transport enables Winston logging to the console.
```
Comment (mrice32) [inline packages/logger/src/logger/ConsoleTransport.ts:1]: Was any content of these files modified? Just want to make sure we specifically review anything that changed!

**Review by mrice32 — APPROVED** [review]
I think moving the logger out is a great idea! Most packages outside the monorepo only use financial-templates-lib for the logger, so this will help reduce the scope of that import.

Should we also drop this code from financial-templates-lib and replace it with a re-export of this code?

**Comment by chrismaree** [comment]
> Should we also drop this code from financial-templates-lib and replace it with a re-export of this code?
I gave the same feedback offline :) I think @md0x is going to do such.

File: packages/logger/package.json (hunk 1 above)
Comment (chrismaree) [reply packages/logger/package.json:27]: agreed! it seems like more than we need! 

Also, this is still dependent on `@uma/common` which means that using this package will drag with it ethers 5.

File: packages/logger/src/logger/ConsoleTransport.ts (hunk 3 above)
Comment (md0x) [reply packages/logger/src/logger/ConsoleTransport.ts]: No, I've just copy pasted files. None of them have been modified

**Pull request merged**